core/templates/dev/head/pages/admin/misc_tab/AdminMiscTabDirective.js

```diff
@@ -76,6 +79,51 @@ oppia.directive('adminMiscTab', [
         $scope.downloadTopicSimilaritiesFile = function() {
           $window.location.href = ADMIN_TOPICS_CSV_DOWNLOAD_HANDLER_URL;
         };
+
+        var setDataExtractionQueryStatusMessage = function(message) {
+          $scope.showDataExtractionQueryStatus = true;
+          $scope.dataExtractionQueryStatusMessage = message;
+        };
+
+        $scope.submitQuery = function() {
+          var STATUS_PENDING = (
+            'Data extraction query has been submitted. Please wait.');
+          var STATUS_FINISHED = 'Loading the extracted data ...';
+          var STATUS_FAILED = 'Error, ';
+
+          var data = {
+            exp_id: $scope.exp_id,
+            exp_version: $scope.exp_version,
+            state_name: $scope.state_name,
+            num_answers: $scope.num_answers
+          };
+
+          setDataExtractionQueryStatusMessage(STATUS_PENDING);
+
+          $http.post(DATA_EXTRACTION_QUERY_HANDLER_URL, data).then(
+            function(response) {
+              var data = response.data;
+              setDataExtractionQueryStatusMessage(STATUS_FINISHED);
+              var extractedAnswers = JSON.stringify(data.data);
+              var dataURL = (
+                'data:application/json;base64,' + btoa(extractedAnswers));
+              var downloadURL = (
+                '<a href="' + dataURL +
+                '" download="data.json"> download </a>');
+              $window.location.href = (
+                'data:text/html;base64,' + btoa(downloadURL));
+            }, function(response) {
+            setDataExtractionQueryStatusMessage(
+              STATUS_FAILED + response.data.error);
+          });
+        };
+
+        $scope.resetForm = function() {
+          $scope.exp_id = null;
```

Comment:
I guess you are right. `null` isn't a string but '' is actually an empty string and we can invoke string functions on it.